src/Controller/ArticleController.php

```diff
@@ -75,15 +75,20 @@ public function add(): string
         $categoryManager = new CategoryManager();
         $categories = $categoryManager->getAllCategory();
 
-        $this->uploadFile();
+        if ($_POST) {
+            $newArticle = $_POST;
+            $imageArticle = $this->uploadFile();
+            $newArticle['imgSrc'] = $imageArticle;
 
+            (new ArticleManager())->createArticle($newArticle);
+        }
         return $this->twig->render('Admin/Article/add.html.twig', [
             'authors' => $authors,
             'categories' => $categories
         ]);
     }
 
-    public function uploadFile(): void
+    public function uploadFile(): string
     {
         if ($_SERVER['REQUEST_METHOD'] === 'POST') {
```

Comment:
you could avoid to check if the method is POST by including your uploadFile method directly into the condition of you add method